tests/server/test_purge_on_delete.py

```diff
@@ -257,7 +262,7 @@ async def test_purge_on_delete_compile_failed(client, server, clienthelper, envi
         version=version,
         resources=[],
         unknowns=[{"parameter": "a", "source": "b"}],
-        version_info={const.EXPORT_META_DATA: {const.META_DATA_COMPILE_STATE: const.Compilestate.failed}},
+        version_info={const.EXPORT_META_DATA: {const.META_DATA_COMPILE_STATE: const.Compilestate.failed.name}},
```

Comment:
Changing the superclass of `Compilestate` from `Enum` to `IntEnum` meant that the "automatic conversion" of any of its enum is done to its integer value, rather than its string name.  What might it break?

Reply:
The superclass has been changed to `str` and `Enum`, and the values to strings matching the name.  As this is how it was used everywhere in the code.